src/magic.cpp

```diff
@@ -895,6 +906,7 @@ bool spell::is_valid_target( const Creature &caster, const tripoint &p ) const
         valid = valid || ( cr_att == Creature::A_FRIENDLY && is_valid_target( target_ally ) &&
                            p != caster.pos() );
         valid = valid || ( is_valid_target( target_self ) && p == caster.pos() );
+        valid = valid || ( is_valid_target( target_monster_id ) && target_by_monster_id( *cr ) );
```

Comment:
this makes it impossible to specifically say "you can only target hostile monsters with this id" for example

Reply:
Changed the targeting logic, if I understood your suggestions correctly. It should work as you specified